File: clients/typescript-angular/API/Client/UserPreferences.ts
/// <reference path="api.d.ts" />

namespace API.Client {
    'use strict';

    export interface UserPreferences {

        announcementsLastSeen?: date;

        colorTheme?: string;

        currency?: string;

        debug?: boolean;

        disableEmails?: Array<string>;

        hideConfirmDialogs?: Array<string>;

        hideConnectionModal?: boolean;

        hideFromLeaderboard?: boolean;

        hideNameFromLeaderboard?: boolean;

        hideNotifications?: Array<string>;

        hidePhoneConfirm?: boolean;

        locale?: string;

        msgsSeen?: Array<string>;

        orderBookBinning?: number;

        orderBookType?: string;

        orderControlsPlusMinus?: boolean;

        sounds?: Array<string>;

        strictIPCheck?: boolean;

        strictTimeout?: boolean;

        tickerGroup?: string;

        tickerPinned?: boolean;

        tradeLayout?: string;
    }

}

File: clients/scala/src/main/scala/io/swagger/client/model/UserPreferences.scala
package io.swagger.client.model

import java.util.Date



case class UserPreferences (
  announcementsLastSeen: Date,
  colorTheme: String,
  currency: String,
  debug: Boolean,
  disableEmails: List[String],
  hideConfirmDialogs: List[String],
  hideConnectionModal: Boolean,
  hideFromLeaderboard: Boolean,
  hideNameFromLeaderboard: Boolean,
  hideNotifications: List[String],
  hidePhoneConfirm: Boolean,
  locale: String,
  msgsSeen: List[String],
  orderBookBinning: Number,
  orderBookType: String,
  orderControlsPlusMinus: Boolean,
  sounds: List[String],
  strictIPCheck: Boolean,
  strictTimeout: Boolean,
  tickerGroup: String,
  tickerPinned: Boolean,
  tradeLayout: String)
  
